routes/tasks.js

```diff
@@ -45,21 +44,24 @@ router
         constants.stringLimits["textBody"]
       );
       utils.validateDate(dateAddedTo, "dateAddedTo");
-      utils.validateDate(dateDueOn, "dateDueOn");
       utils.validatePriority(priority);
       utils.validateStringInputWithMaxLength(
         tag,
         "tag",
         constants.stringLimits["tag"]
       );
+      if (typeof checked === "undefined") {
+        checked = false;
+      }
+      utils.validateBooleanInput(checked, "checked");
```

Comment:
this looks good can be used for the other review comments above

Reply:
changed